app/Http/Controllers/registrationController.php

```diff
@@ -3,11 +3,98 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use App\Models\Users;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Redis;
+use App\Models\tran;
+use Illuminate\Support\Facades\Redirect;
 
 class registrationController extends Controller
 {
-    public function index()
+    public function registerShow()
     {
         return view('form');
     }
+    public function loginShow()
+    {
+        return view('login');
+    }
+    public function register()
+    {
+        return redirect('welcome');
+        
+    }
+    public function view(){
+        $people = Users::latest('updated_at')->get();
+        $data = compact('people');
+        return view('customer-view')->with($data);
+    }
+    public function log(Request $request)
+    {
+        $request->validate([
+            'email' => 'required | email',
+            'pass' => 'required | min:5',
+        ]);
+
```

Comment:
It is recommend to write minimum code on Controllers, as the Controllers are most frequent files that developer access, so overall line of code needs to less in no.s .. As validate function can be written in separate request class, we can remove this code from here.

Additional reference -> https://medium.com/@kamerk22/the-smart-way-to-handle-request-validation-in-laravel-5e8886279271